### PR TITLE
[kostal] Add channels (#1021)

### PR DESCRIPTION
**Pull request by cschneider**
Signed-off-by: Christian Schneider <cschneid@adobe.com> (github: cschneider)

Integrated issue from
#1021

Adding some more channels

Tested in Paper UI.

### CODE REVIEW

File: addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -38,7 +48,77 @@
 			</parameter>
 		</config-description>
 	</thing-type>
-
+	
+	<channel-type id="str1Voltage">
+	   <item-type>Number</item-type>
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml]: As the binding already uses UoM we should change the item types to deliver a [QuantityType](https://www.openhab.org/docs/concepts/units-of-measurement.html#list-of-units).

```suggestion
	   <item-type>Number:ElectricPotential</item-type>
```

File: addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -38,7 +48,77 @@
...
+	   <label>Str1 Voltage</label>
+	   <category>Energy</category>
+	   <state readOnly="true" pattern="%.2f V" />
+    </channel-type>
+    
+	<channel-type id="str1Current">
+		<item-type>Number</item-type>
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml]:
```suggestion
		<item-type>Number:ElectricCurrent</item-type>
```

File: addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -38,7 +48,77 @@
...
+	
+	<channel-type id="l1Voltage">
+		<item-type>Number</item-type>
+		<label>L1 Voltage</label>
+		<category>Energy</category>
+		<state readOnly="true" pattern="%.2f V" />
+	</channel-type>
+	
+	<channel-type id="l1Power">
+		<item-type>Number</item-type>
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml]:
```suggestion
		<item-type>Number:Power</item-type>
```

File: addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml [hunk 4]
```diff
@@ -38,7 +48,77 @@
 			</parameter>
 		</config-description>
 	</thing-type>
-
+	
+	<channel-type id="str1Voltage">
+	   <item-type>Number</item-type>
+	   <label>Str1 Voltage</label>
+	   <category>Energy</category>
+	   <state readOnly="true" pattern="%.2f V" />
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml]: We can use a placeholder for the unit.
```suggestion
	   <state readOnly="true" pattern="%.2f %unit%" />
```

File: addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml [hunk 5]
```diff
@@ -12,6 +12,16 @@
 			<channel typeId="totalEnergy" id="totalEnergy"></channel>
 			<channel typeId="dayEnergy" id="dayEnergy"></channel>
 			<channel typeId="status" id="status"></channel>
+			<channel typeId="str1Voltage" id="str1Voltage" ></channel>
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kostalinverter/ESH-INF/thing/thing-types.xml]: Optional: If you are touching it anyways you can shorten the syntax to:
```
<channel typeId="str1Voltage" id="str1Voltage" />
```

**Comment by thopiekar** [comment]
Awesome 👍

**Comment by cweitkamp** [comment]
@cschneider Thanks for the update. Did you catch all my comments?

Please ping one of us next time. We do not receive a notification for forced pushes.

**Comment by cschneider** [comment]
I hope I got all comments.

**Comment by cweitkamp** [comment]
Can you please double check the new voltage channels for item type `Number:ElectricPotential`.

**Comment by cschneider** [comment]
I changed to ElectricPotential and also added the channels to docs.

**Pull request merged**